Source/OxyPlot/Axes/Rendering/NumericColorAxisRenderer.cs

```diff
@@ -0,0 +1,116 @@
+                var reverse = axisLength < 0;
+                axisLength = Math.Abs(axisLength);
+
+                if (axis.IsHorizontal())
+                {
+                    var colorAxisImage = this.GenerateColorAxisImage(axis, reverse);
+                    this.RenderContext.DrawImage(colorAxisImage, this.left, this.top, axisLength, this.size, 1, true);
+                }
+                else
+                {
+                    var colorAxisImage = this.GenerateColorAxisImage(axis, reverse);
+                    this.RenderContext.DrawImage(colorAxisImage, this.left, this.top, this.size, axisLength, 1, true);
+                }
+            }
+            else
+            {
+                for (var i = 0; i < axis.Palette.Colors.Count; i++)
+                {
+                    var ylow = this.Transform(axis, axis.GetLowValue(i));
+                    var yhigh = this.Transform(axis, axis.GetHighValue(i));
+                    this.DrawColorRect(axis, ylow, yhigh, axis.Palette.Colors[i]);
+                }
+
+                double highLowLength = 10;
+                if (axis.IsHorizontal())
+                {
+                    highLowLength *= -1;
+                }
+
+                if (!axis.LowColor.IsUndefined())
+                {
+                    this.DrawColorRect(axis, this.minScreenPosition, this.minScreenPosition + highLowLength, axis.LowColor);
+                }
+
+                if (!axis.HighColor.IsUndefined())
+                {
+                    this.DrawColorRect(axis, this.maxScreenPosition, this.maxScreenPosition - highLowLength, axis.HighColor);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generates the image used to render the color axis.
+        /// </summary>
+        /// <param name="axis">The color axis.</param>
+        /// <param name="reverse">Reverse the colors if set to <c>true</c>.</param>
+        /// <returns>An <see cref="OxyImage" /> used to render the color axis.</returns>
+        private OxyImage GenerateColorAxisImage(T axis, bool reverse)
+        {
+            var n = axis.Palette.Colors.Count;
+            var buffer = axis.IsHorizontal() ? new OxyColor[n, 1] : new OxyColor[1, n];
+            for (var i = 0; i < n; i++)
+            {
+                var color = axis.Palette.Colors[i];
+                var i2 = reverse ? n - 1 - i : i;
+                if (axis.IsHorizontal())
+                {
+                    buffer[i2, 0] = color;
+                }
+                else
+                {
+                    buffer[0, i2] = color;
+                }
+            }
+
+            return OxyImage.Create(buffer, ImageFormat.Png);
+        }
+
+        /// <summary>
+        /// Transforms a value to a screen position. We don't use the regular Transform functions of the axis here, as the color block should always be drawn with linear scaling.
```

Comment:
Not sure this makes sense; will have to come back to this

Reply:
Even for `LogarithmicColorAxis`, we want the the color transition to be drawn in linear space, as it would not be very readable in logarithmic.
The transformation to log space happens [elsewhere](https://github.com/oxyplot/oxyplot/pull/2049/files/25962e7e7e12de834d09282004a0fbee3ad38f5a#diff-4655df4176a03ff37dff77dc3d2bbfe2c67a1c17c83c2c56bf4172547e3fda9dR85). This is why we have to use this linear transformation here. Does this make sense?

Reply:
It certainly sounds correct, but something felt fishy to me so I'd like to take another look when I have a bit more time. Happy for this to be merged in the interim.